src/main/java/org/mifos/connector/mockpaymentschema/service/AirtelMockService.java

```diff
@@ -0,0 +1,158 @@
+    }
+
+    public ResponseEntity<AirtelPaymentResponseDTO> initiateTransaction(AirtelPaymentRequestDTO airtelPaymentRequestDTO) {
+        String message;
+        String status;
+        String code;
+        String responseCode;
+        boolean success;
+        HttpStatus httpStatus;
+
+        String msisdn = airtelPaymentRequestDTO.getSubscriber().getMsisdn();
+        if (msisdn.equals(msisdnSuccessful)) {
+            message = "";
+            status = TransferStatus.SUCCESSFUL.name();
+            code = "200";
+            responseCode = successResponseCode;
+            success = true;
+            httpStatus = HttpStatus.OK;
+        }
+
+        else if (msisdn.equals(msisdnFailed)) {
+            message = "";
+            status = TransferStatus.FAILED.name();
+            code = "400";
+            responseCode = failedResponseCode;
+            success = false;
+            httpStatus = HttpStatus.BAD_REQUEST;
+        }
+
+        else {
+            message = "";
+            status = TransferStatus.IN_PROGRESS.name();
+            code = "202";
+            responseCode = pendingResponseCode;
+            success = true;
+            httpStatus = HttpStatus.ACCEPTED;
+        }
+
+        return airtelPaymentRespone(message, status, code, responseCode, success, httpStatus);
+
+    }
+
+    private ResponseEntity<AirtelEnquiryResponseDTO> airtelEnquiryResponse(String airtelMoneyId, String transactionId, String message,
+            String status, String code, String responseCode, Boolean success, HttpStatus httpStatus) {
+        String id = transactionId;
+
+        AirtelEnquiryResponseDataTransactionDTO airtelEnquiryResponseDataTransactionDTO = new AirtelEnquiryResponseDataTransactionDTO(
+                airtelMoneyId, id, message, status);
+        AirtelEnquiryResponseDataDTO airtelResponseDataDTO = new AirtelEnquiryResponseDataDTO(airtelEnquiryResponseDataTransactionDTO);
+
+        String resultCode = "ESB000010";
+
+        AirtelResponseStatusDTO airtelResponseStatusDTO = new AirtelResponseStatusDTO(code, message, resultCode, responseCode, success);
+
+        AirtelEnquiryResponseDTO responseEntity = new AirtelEnquiryResponseDTO(airtelResponseDataDTO, airtelResponseStatusDTO);
+        return ResponseEntity.status(httpStatus).body(responseEntity);
+
+    }
+
+    private ResponseEntity<AirtelPaymentResponseDTO> airtelPaymentRespone(String message, String status, String code, String responseCode,
+            Boolean success, HttpStatus httpStatus) {
+        boolean id = false;
+
+        AirtelPaymentResponseDataTransactionDTO airtelPaymentResponseDataTransactionDTO = new AirtelPaymentResponseDataTransactionDTO(id,
+                status);
+        AirtelPaymentResponseDataDTO airtelResponseDataDTO = new AirtelPaymentResponseDataDTO(airtelPaymentResponseDataTransactionDTO);
+
+        String resultCode = "ESB000010";
```

Comment:
instead of hardcoding pick from application.yml or use a global variable

Reply:
Done.